src/test/test_group_np_cli.c

```diff
@@ -0,0 +1,179 @@
+ *     this software are asked, but not required, to acknowledge that it was
+ *     developed by Intel Corporation and credit the contributors.
+ *
+ * 5. Neither the name of Intel Corporation, nor the name of any Contributor
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This is a simple example of cart rpc group test based on crt API.
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdint.h>
+#include <assert.h>
+#include <getopt.h>
+#include <semaphore.h>
+
+#include "tests_common.h"
+#include "crt_fake_events.h"
+#include "test_group_rpc.h"
+#include "test_group_np_common.h"
+
+void
+test_run(void)
+{
+        crt_group_t             *grp = NULL;
+        d_rank_list_t           *rank_list = NULL;
+        d_rank_t                rank;
+	int			tag;
+        crt_endpoint_t                   server_ep = {0};
+        crt_rpc_t                       *rpc_req = NULL;
+	int		i;
+	int		rc = 0;
+
+	tc_cli_start_basic(test_g.t_local_group_name,
+			   test_g.t_remote_group_name,
+			   &grp, &rank_list, &test_g.t_crt_ctx[0],
+			   &test_g.t_tid[0], test_g.t_srv_ctx_num);
+
+        rc = sem_init(&test_g.t_token_to_proceed, 0, 0);
+        D_ASSERTF(rc == 0, "sem_init() failed.\n");
+
+	/* register RPCs */
+	rc = crt_proto_register(&my_proto_fmt_test_group2);
+	D_ASSERTF(rc == 0, "crt_proto_register() failed. rc: %d\n",
+			rc);
+
+        sleep(2);
+
+        rc = wait_for_ranks(test_g.t_crt_ctx[0], grp, rank_list,
+			    test_g.t_srv_ctx_num - 1, test_g.t_srv_ctx_num,
+			    5, 150);
+        if (rc != 0) {
+                D_ERROR("wait_for_ranks() failed; rc=%d\n", rc);
+                assert(0);
+        }
+
+	test_g.t_fault_attr_1000 = d_fault_attr_lookup(1000);
+	test_g.t_fault_attr_5000 = d_fault_attr_lookup(5000);
+
+        if (!test_g.t_shut_only) {
+
+        	for (i = 0; i < rank_list->rl_nr; i++) {
+
+                	rank = rank_list->rl_ranks[i];
+
+			for (tag = 0; tag < test_g.t_srv_ctx_num; tag++) {
+                		DBG_PRINT("Sending rpc to %d:%d\n", rank, tag);
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs